src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,42 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int ARRAY_LENGTH = 10;
+    private Object[] keys;
+    private Object[] values;
+    private int size;
+
+    public StorageImpl() {
+        this.keys = new Object[ARRAY_LENGTH];
+        this.values = new Object[ARRAY_LENGTH];
+        this.size = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < size; i++) {
+            if (key == null && keys[i] == null || key != null && key.equals(keys[i])) {
+                values[i] = value;
+                return;
+            }
+        }
+        keys[size] = key;
+        values[size] = value;
+        size++;
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < size; i++) {
+            if (key == null && keys[i] == null || key != null && key.equals(keys[i])) {
```

Comment:
you can simplify this conditional
```suggestion
            if (key == keys[i] || key != null && key.equals(keys[i])) {
```